mempool.go

```diff
@@ -610,12 +611,55 @@ func (mp *txMemPool) removeTransaction(tx *btcutil.Tx) {
 	// Remove the transaction and mark the referenced outpoints as unspent
 	// by the pool.
 	if txDesc, exists := mp.pool[*txHash]; exists {
+		if cfg.AddrIndex {
+			mp.removeTransactionFromAddrIndex(tx)
+		}
+
 		for _, txIn := range txDesc.Tx.MsgTx().TxIn {
 			delete(mp.outpoints, txIn.PreviousOutPoint)
 		}
 		delete(mp.pool, *txHash)
 		mp.lastUpdated = time.Now()
 	}
+
+}
+
+// removeTransactionFromAddrIndex removes the passed transaction from our
+// address based index.
```

Comment:
Done. I've added similar documentation to related functions that were added to the mempool in this PR.